lib/streamlit/runtime/websocket_session_manager.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Optional, TYPE_CHECKING
+
+from typing_extensions import Final
+
+from streamlit.logger import get_logger
+from streamlit.watcher import LocalSourcesWatcher
+from .app_session import AppSession
+from .session_data import SessionData
+from .session_manager import SessionClient, SessionInfo, SessionManager, SessionStorage
+
+if TYPE_CHECKING:
+    from .runtime import Runtime
+
+LOGGER: Final = get_logger(__name__)
+
+
+class WebsocketSessionManager(SessionManager):
+    def __init__(self, session_storage: SessionStorage, runtime: "Runtime") -> None:
+        # NOTE: We intentionally don't do anything with session_storage for now as we
+        # will initially implement WebsocketSessionManager so the current runtime
+        # session handling behavior is unchanged.
+        #
+        # A concrete session_storage class will be implemented and used in subsequent
+        # changes when we add session reuse on a websocket reconnect.
+        #
+        # Since we aren't worrying about session reconnects yet, we don't need to
+        # differentiate between active and inactive sessions and can provide the minimal
+        # implementation of a SessionManager below.
+
+        self._runtime = runtime
+
+        # Mapping of AppSession.id -> SessionInfo.
+        self._session_info_by_id: Dict[str, SessionInfo] = {}
+
+    def connect_session(
+        self,
+        client: SessionClient,
+        user_info: Dict[str, Optional[str]],
+        existing_session_id: Optional[str] = None,  # unused for now
+    ) -> str:
+        runtime = self._runtime
+
+        session = AppSession(
+            event_loop=runtime.get_async_objs().eventloop,
```

Comment:
Ah - if `get_async_objs` is being made public just for its eventloop accessor, let's just pass `eventloop` to the connect_session function?

Alternately: I wonder if the eventloop passing is itself a bit of a mistake in the original Runtime code. These functions are required to execute on the eventloop thread, so a better pattern might be for e.g. the AppSession constructor to do `self._event_loop = asyncio.get_running_loop()` (which will assert if there is no running loop, which is what we want)